pkg/auth/types.go

```diff
@@ -0,0 +1,34 @@
+/*
+Copyright 2019-2020 vChain, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package auth
+
+import "context"
+
+type TokenAuth struct {
+	Token string
+}
+
+func (t TokenAuth) GetRequestMetadata(ctx context.Context, in ...string) (map[string]string, error) {
+	return map[string]string{
+		"authorization": "Bearer " + t.Token,
+	}, nil
+}
+
+func (TokenAuth) RequireTransportSecurity() bool {
+	// TODO OGG: set this to true, unless config says is ok to return false (i.e. for testing)
```

Comment:
You forgot to switch this?

Reply:
@mmeloni Not really forgot, but it is a bit tricky to set it to `true` since in the case when both client and server run on the same machine authentication should work even without MTLS.